src/components/flatSelect/FlatSelectReducers.ts

```diff
@@ -0,0 +1,54 @@
+import { IReduxAction } from '../../utils/ReduxUtils';
+import * as _ from 'underscore';
+import { FlatSelectActions, IFlatSelectActionPayload } from './FlatSelectActions';
+import { IFlatSelectOptionProps } from './FlatSelectOption';
+
+export interface IFlatSelectState {
+  id: string;
+  selectedOption?: IFlatSelectOptionProps;
+}
+
+export const flatSelectInitialState: IFlatSelectState = { id: undefined, selectedOption: undefined };
+export const flatSelectsInitialState: IFlatSelectState[] = [];
+
+export const flatSelectReducer = (state: IFlatSelectState = flatSelectInitialState,
+  action: (IReduxAction<IFlatSelectActionPayload>)): IFlatSelectState => {
+  switch (action.type) {
+    case FlatSelectActions.select:
+      if (state.id !== action.payload.id) {
+        return state;
+      }
```

Comment:
The id check is duplicated in `flatSelectsReducer`. I don't know which one you should keep but I don't think you need both

Reply:
removed